audiences/app/models/audiences/external_user.rb

```diff
@@ -13,10 +13,28 @@ class ExternalUser < ApplicationRecord
     def self.fetch(external_ids, count: 100)
       return [] unless external_ids.any?
 
-      Array(external_ids).in_groups_of(count, false).flat_map do |ids|
-        filter = Array(ids).map { "externalId eq #{_1}" }.join(" OR ")
-        Audiences::Scim.resource(:Users).all(count: count, filter: filter).to_a
+      # Standardize the ids to an array of strings
+      ids = Array(external_ids).map(&:to_s)
+
+      # The local database is used as a cache for SCIM users, because this is significantly faster than querying SCIM
+      stored_users = Audiences::ExternalUser.where(user_id: ids).to_a
```

Comment:
@gregblake Have you considered how we'd mark local data stale so we know when to refresh it from SCIM? Would something like following work to ensure we refresh the data if it's stale?
```suggestion
      stored_users = Audiences::ExternalUser.where(user_id: ids, updated_at: 15.minutes.ago..DateTime.current).to_a
```